### PR TITLE
Detailed return types for `str_pad()` and `str_repeat()`

### PR DESCRIPTION
**Pull request by weirdan**
Fixes vimeo/psalm#6049

### CODE REVIEW

File: stubs/CoreGenericFunctions.phpstub [hunk 1]
```diff
@@ -762,13 +762,26 @@ function str_ireplace($search, $replace, $subject, &$count = null) {}
...
+ * @return ($input is non-empty-string ? non-empty-string : ($pad_length is positive-int ? non-empty-string: string))
+ *
  * @psalm-flow ($input, $pad_string) -> return
  */
 function str_pad(string $input, int $pad_length, $pad_string = '', int $pad_type = STR_PAD_RIGHT): string {}
 
 /**
  * @psalm-pure
  *
+ * @todo update $multiplier to be `0|positive-int`
```
Comment (orklah) [inline stubs/CoreGenericFunctions.phpstub:774]: Did you left this on purpose?

**Comment by ciaranmcnulty** [comment]
str_pad enforces that the pad_string is non-empty - is that something that should be captured as a param?

File: stubs/CoreGenericFunctions.phpstub (hunk 1 above)
Comment (weirdan) [reply stubs/CoreGenericFunctions.phpstub:774]: Yes. `str_repeat()` throws warnings when given negative int, but Psalm was emitting too many false positive coercion issues (int -> positive int) even on its own codebase.

**Comment by weirdan** [comment]
>  is that something that should be captured as a param?

Yeah, it needs to be typed as `non-empty-string` then.

**Pull request merged**